### PR TITLE
Ria 2220 save for later

### PR DESCRIPTION
**Pull request by chrisgrimble**
Added save for later functionality to the appeal application
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/RIA-2220

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[X] No
```

### CODE REVIEW

File: test/e2e-test/pages/home-office-details/home-office-letter-sent.ts [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+const config = require('config');
+const { fillInDate } = require('../helper-functions');
+const cache = require('memory-cache');
+
+const testUrl = config.get('testUrl');
+
+module.exports = {
+  homeOfficeLetterSent(I) {
+    Given('I am on the home office letter sent page', async () => {
+      I.amOnPage(testUrl + paths.homeOffice.letterSent);
```
Comment (t0nin0s) [inline test/e2e-test/pages/home-office-details/home-office-letter-sent.ts]: Shall we make sure there is no error when navigating to a page? Something like checking page title?

File: app/app.ts [hunk 2]
```diff
@@ -11,6 +11,8 @@ import { logErrorMiddleware, logRequestMiddleware } from './middleware/logger';
 import { router } from './routes';
 import { setupSession } from './session';
 
+const cache = require('memory-cache');
```
[the remark this answers is not in the document]
Comment (chrisgrimble) [reply app/app.ts]: Nope removed

File: app/controllers/home-office-details.ts [hunk 3]
```diff
@@ -90,7 +96,7 @@ function postDateLetterSent(updateAppealService: UpdateAppealService) {
 
       }
       req.session.appeal.application.isAppealLate = true;
-      res.redirect(paths.homeOffice.appealLate);
+      res.redirect(getNextPage(req.body, paths.homeOffice.appealLate));
```
[the remark this answers is not in the document]
Comment (chrisgrimble) [reply app/controllers/home-office-details.ts]: Nope getConditionalRedirectUrl will send you to the check your answers page if that's where you came from. We always want to go to the appeal late page if your appeal is late. This is to be changed with a new story anyway.

File: test/e2e-test/pages/common.ts [hunk 4]
```diff
@@ -1,12 +1,170 @@
+import dateFormat from 'dateformat';
```
[the remark this answers is not in the document]
Comment (chrisgrimble) [reply test/e2e-test/pages/common.ts]: Done

File: test/e2e-test/pages/helper-functions.ts [hunk 5]
```diff
@@ -47,6 +47,11 @@ async function signInHelper() {
   I.click('.button');
 }
 
+function signInForUser(email: string) {
+  I.fillField('#username', email);
+  I.click('.button');
```
[the remark this answers is not in the document]
Comment (chrisgrimble) [reply test/e2e-test/pages/helper-functions.ts]: Yeh have changed

**Pull request merged**